src/virtualenv/discovery/py_info.py

```diff
@@ -237,7 +237,7 @@ def satisfies(self, spec, impl_must_match):
                 return False
 
         if impl_must_match:
-            if spec.implementation is not None and spec.implementation != self.implementation:
+            if spec.implementation is not None and spec.implementation.lower() != self.implementation.lower():
```

Comment:
You'll want to do the same at https://github.com/pypa/virtualenv/blob/master/src/virtualenv/discovery/py_spec.py#L101, and also add test for that too 👍